File: java/leetcode_grind/Day730.java
package leetcode_grind;

public class Day730 {
    // https://leetcode.com/problems/find-the-power-of-k-size-subarrays-i/description/?envType=daily-question&envId=2024-11-16
    static class Solution1 {
        public int[] resultsArray(int[] nums, int k) {
            int length = nums.length;
            int[] result = new int[length - k + 1];

            for (int start = 0; start <= length - k; start++) {
                boolean isConsecutiveAndSorted = true;

                for (int index = start; index < start + k - 1; index++) {
                    if (nums[index + 1] != nums[index] + 1) {
                        isConsecutiveAndSorted = false;
                        break;
                    }
                }

                if (isConsecutiveAndSorted) {
                    result[start] = nums[start + k - 1];
                } else {
                    result[start] = -1;
                }
            }

            return result;
        }
    }

}

File: src/leetcode_grind/day_730.rs
// https://leetcode.com/problems/find-the-power-of-k-size-subarrays-i/description/?envType=daily-question&envId=2024-11-16
pub fn results_array(nums: Vec<i32>, k: i32) -> Vec<i32> {
    if k == 1 {
        return nums;
    }

    let length = nums.len();
    let mut result = vec![-1; length - k as usize + 1];

    let mut consecutive_count = 1;

    for index in 0..length - 1 {
        if (nums[index] + 1 == nums[index + 1]) {
            consecutive_count += 1;
        } else {
            consecutive_count = 1;
        }

        if consecutive_count >= k {
            result[index - k as usize + 2] = nums[index + 1];
        }
    }

    result
}
